providers/os/resources/vulnmgmt.go

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package resources
+
+import (
+	"errors"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"go.mondoo.com/cnquery/v9/llx"
+	"go.mondoo.com/cnquery/v9/providers-sdk/v1/plugin"
+	"go.mondoo.com/cnquery/v9/providers-sdk/v1/resources"
+	"go.mondoo.com/cnquery/v9/providers-sdk/v1/upstream/gql"
+	"go.mondoo.com/cnquery/v9/providers/os/connection/shared"
+	mondoogql "go.mondoo.com/mondoo-go"
+)
+
+func (v *mqlVulnmgmt) lastAssessment() (*time.Time, error) {
+	mcc := v.MqlRuntime.Upstream
+	if mcc == nil || mcc.ApiEndpoint == "" {
+		return nil, resources.MissingUpstreamError{}
+	}
+
+	// get new gql client
+	mondooClient, err := gql.NewClient(mcc.UpstreamConfig, mcc.HttpClient)
```

Comment:
you are creating a new GQL client in multiple functions. Isn't it better to make the GQL client part of the connection or part of the `mqlVulnmgmt` instead?

Reply:
Nice idea.
I added it to a cache, so we can re-use it, but do not need to create it upfront.